src/control/SConscript

```diff
@@ -46,20 +47,32 @@ def gen_build_id():
     return '0x' + buildid.decode()
 
 
+def get_build_info():
+    """Attempt to retrieve commit/tag details from the build environment."""
+    try:
+        cmd = subprocess.run(['git', 'describe', '--tags', '--dirty', '--always'],
```

Comment:
It would be useful if we got all build info here and and spit it to a file and exit when --build-info=write is set and read it from a file if --build-info=read and only automatically get it if --build-info=get (default) or something like that